sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -380,29 +404,44 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
      */
     public Mono<MsalToken> authenticateWithUsernamePassword(TokenRequestContext request,
                                                             String username, String password) {
-        return Mono.fromFuture(publicClientApplication.acquireToken(
+        return Mono.fromFuture(() -> getPublicClientApplication(false).acquireToken(
```

Comment:
Do we ever want this future to be cancellable? If yes, we will need to add a `doFinally` operator that will check for cancelled to cancel the future.

Reply:
Synced offline - due to the small cost of the getToken() call we will skip the hassle to cancel the future here

Reply:
`acquireToken` here will most likely make an HttpRequest, would this normally be cancellable in other service clients? If so we should probably do the same here.

Reply:
As discussed, there's nothing much to gain to chain doOnCancel signal to CompletableFuture, as it gets triggered right away and not when subscribed to.